libraries/AP_NavEKF2/AP_NavEKF2_VehicleStatus.cpp

```diff
@@ -266,7 +266,7 @@ void NavEKF2_core::calcGpsGoodForFlight(void)
 
     // get the receivers reported speed accuracy
     float gpsSpdAccRaw;
-    if (!AP::gps().speed_accuracy(gpsSpdAccRaw)) {
+    if (AP::gps().status() < AP_GPS::GPS_OK_FIX_2D ||!AP::gps().speed_accuracy(gpsSpdAccRaw)) {
```

Comment:
I'm not sure this is better. Effectively this pins both the `sAccFilterState1` and `sAccFilterState2` at 0 when you don't have a GPS fix, which then leads to ignoring the speed accuracy requirements if the GPS notionally has them.

Reply:
why would the GPS have them?

Reply:
It isn't obvious externally but the GPS drivers actually use the flag of `have_speed_accuracy` to indicate that they are receiving the data stream with the accuracy fields, so we can be at no fix, but we will have speed accuracy information we'd want to be using (and some receivers will pin this value at an insanely high number until after we get fix, which before would ensure we wouldn't get through this check). Now not all receivers actually pin the accuracy high, so I think something that says if we get an accuracy but aren't at least 2D fix then we always fail the speed check is actually a good improvement, but what's here removes a level of protection from some systems.

Reply:
You're the GPS maintainer, so you'll get the final say, but just to clarify - lock status is irrelevant to the speed accuracy accessor (which does things behind the scenes, possibly even checking status itself). If this is the case then it's Ok to remove this status check